endpoints/openrtb2/auction_test.go

```diff
@@ -227,16 +229,74 @@ func TestBlacklistRequests(t *testing.T) {
 	tests.assert(t)
 }
 
+// TestRejectAccountRequired asserts we return a 400 code on a request that comes with no user id nor app id
+// if the `AccountRequired` field in the `config.Configuration` structure is set to true
+func TestRejectAccountRequired(t *testing.T) {
+	tests := map[string]*getResponseFromDirectory{
+		"8.1) Not required and not provided. Since not provided, not blacklisted": {
+			dir:           "sample-requests/account-required",
+			file:          "no-acct.json",
+			payloadGetter: getRequestPayload,
+			messageGetter: nilReturner,
+			expectedCode:  http.StatusOK,
+			accountReq:    false,
+		},
+		"8.2) Required, not provided and in consequence, not blacklisted": {
+			dir:           "sample-requests/account-required",
+			file:          "no-acct.json",
+			payloadGetter: getRequestPayload,
+			messageGetter: getMessage,
+			expectedCode:  http.StatusBadRequest,
+			accountReq:    true,
+		},
+		"8.3) Required, provided and not blacklisted": {
+			dir:           "sample-requests/account-required",
+			file:          "with-acct.json",
+			payloadGetter: getRequestPayload,
+			messageGetter: nilReturner,
+			expectedCode:  http.StatusOK,
+			aliased:       true,
+			accountReq:    true,
+		},
+		"8.4) Required, provided and blacklisted": {
+			dir:           "sample-requests/blacklisted",
+			file:          "blacklisted-acct.json",
+			payloadGetter: getRequestPayload,
+			messageGetter: getMessage,
+			expectedCode:  http.StatusBadRequest,
```

Comment:
Shouldn't the expected response code here be `http. StatusServiceUnavailable`?

Reply:
Yes, there was a bug in the code. Corrected